Variable Passing: int

Hello gang!

I believe that there might be an issue reading integers from one file when they've been included in another.

For instance:
foo.yaml:
owner-id: 1001

bar.yaml:
!include: foo.yaml
owner-id: {owner-id}
--or
owner-id: "{owner-id:d}"

So far, I haven't been able to get this to return an int to the calling document. If explicitly casting to an int:

bar.yaml:
!include: foo.yaml
owner-id: !!int "{owner-id:d}"

The following error is produced: _ValueError: invalid literal for int() with base 10: '{testid'_
because the cast is happening before the variable is returned.

If I have an incorrect implementation somewhere, please let me know! Thanks!

Hello kind folk at Tavern!

It seems like there has been a resurgence in the issue of casting ints using the "!int '{value:d}'" syntax. This is running version tavern-0.9.2 (which is also available in the logs).

Might you all know why this is happening again? Something to note, is in test_regression_get_collection.tavern.yaml this line:
`collection-url: "/collections/{createdCollectionID:d}"`
works without issue, however,
`collection-id: !int '{createdCollectionID:d}'`
receives an error

Thanks!

Files:
test_regression_get_collection.tavern.yaml:

```
includes:
  - !include CollectionConfig/config.yaml

test_name: Perform a get on collection- 200
stages:
  - name: Get on collection
    request:
      url: "{baseURL:s}/{createdCollectionID:d}"
      method: GET
      headers:
        content-type: application/json
        x-api-key: "{x-api-key:s}"
    response:
      status_code: 200
      body:
        collection-properties:
          collection-id: !int '{createdCollectionID:d}'
          collection-url: "/collections/{createdCollectionID:d}"

```

```
name: config.yaml
description: Collection constants.

variables:
  baseURL: example.com
  x-api-key: lololol
  created-collection-id: 1
  current-collection-name: test_regression_collection2018-04-13 16:21:51.021420
  new-collection-name: test_regression_collection2018-04-13 16:26:17.147835
  createdCollectionID: 1
```



Testing started at 12:38 ...
C:\venv\62-partytown-Regression\Scripts\python.exe C:\Users\bf\.IdeaIC2018.1\config\plugins\python-ce\helpers\pycharm\_jb_pytest_runner.py --path C:/Users/bf/Documents/somewhere/IndividualRepos/62-partytown-Regression/Source/TestCases/test_regression_get_collection.tavern.yaml
Launching py.test with arguments C:/Users/bf/Documents/somewhere/IndividualRepos/62-partytown-Regression/Source/TestCases/test_regression_get_collection.tavern.yaml in C:\Users\bf\Documents\somewhere\IndividualRepos\62-partytown-Regression\Source\TestCases

============================= test session starts =============================
platform win32 -- Python 3.6.4, pytest-3.5.0, py-1.5.3, pluggy-0.6.0
rootdir: C:\Users\bf\Documents\somewhere\IndividualRepos\62-partytown-Regression\Source\TestCases, inifile:
plugins: tavern-0.9.2
test_regression_get_collection.tavern.yaml:None (test_regression_get_collection.tavern.yaml)
C:\venv\62-partytown-Regression\lib\site-packages\tavern\testutils\pytesthook.py:74: in collect
    all_tests = list(yaml.load_all(self.fspath.open(), Loader=IncludeLoader))
C:\venv\62-partytown-Regression\lib\site-packages\yaml\__init__.py:84: in load_all
    yield loader.get_data()
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:31: in get_data
    return self.construct_document(self.get_node())
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:46: in construct_document
    for dummy in generator:
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:398: in construct_yaml_map
    value = self.construct_mapping(node)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:204: in construct_mapping
    return super().construct_mapping(node, deep=deep)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:129: in construct_mapping
    value = self.construct_object(value_node, deep=deep)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:86: in construct_object
    data = constructor(self, node)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:414: in construct_undefined
    node.start_mark)
E   yaml.constructor.ConstructorError: could not determine a constructor for the tag '!int'
E     in "C:\Users\bf\Documents\somewhere\IndividualRepos\62-partytown-Regression\Source\TestCases\test_regression_get_collection.tavern.yaml", line 24, column 26
collected 0 items / 1 errors

=================================== ERRORS ====================================
_________ ERROR collecting test_regression_get_collection.tavern.yaml _________
C:\venv\62-partytown-Regression\lib\site-packages\tavern\testutils\pytesthook.py:74: in collect
    all_tests = list(yaml.load_all(self.fspath.open(), Loader=IncludeLoader))
C:\venv\62-partytown-Regression\lib\site-packages\yaml\__init__.py:84: in load_all
    yield loader.get_data()
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:31: in get_data
    return self.construct_document(self.get_node())
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:46: in construct_document
    for dummy in generator:
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:398: in construct_yaml_map
    value = self.construct_mapping(node)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:204: in construct_mapping
    return super().construct_mapping(node, deep=deep)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:129: in construct_mapping
    value = self.construct_object(value_node, deep=deep)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:86: in construct_object
    data = constructor(self, node)
C:\venv\62-partytown-Regression\lib\site-packages\yaml\constructor.py:414: in construct_undefined
    node.start_mark)
E   yaml.constructor.ConstructorError: could not determine a constructor for the tag '!int'
E     in "C:\Users\bf\Documents\somewhere\IndividualRepos\62-partytown-Regression\Source\TestCases\test_regression_get_collection.tavern.yaml", line 24, column 26
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!
=========================== 1 error in 0.24 seconds ===========================

--- a/tavern/util/loader.py
+++ b/tavern/util/loader.py
@@ -153,10 +153,10 @@ def construct_type_convert(sentinel_type):
 
     def callback(loader, node):
         value = loader.construct_scalar(node)
-        return sentinel_type(value)
+        return sentinel_type.constructor(value)
 
     return callback
 
 
-yaml.loader.Loader.add_constructor("!int", construct_type_convert(IntSentinel))
-yaml.loader.Loader.add_constructor("!float", construct_type_convert(FloatSentinel))
+IncludeLoader.add_constructor("!int", construct_type_convert(IntSentinel))
+IncludeLoader.add_constructor("!float", construct_type_convert(FloatSentinel))

--- a/tests/unit/test_utilities.py
+++ b/tests/unit/test_utilities.py
@@ -1,10 +1,37 @@
+from textwrap import dedent
+
 import pytest
+import yaml
 import copy
 
 from tavern.schemas.extensions import validate_extensions
 from tavern.util import exceptions
-from tavern.util.loader import ANYTHING
+from tavern.util.loader import ANYTHING, IncludeLoader
 from tavern.util.dict_util import deep_dict_merge, check_keys_match_recursive
+
+
+@pytest.fixture(name="test_yaml")
+def fix_test_yaml():
+    text = dedent("""
+    ---
+    test_name: Make sure server doubles number properly
+
+    stages:
+      - name: Make sure number is returned correctly
+        request:
+          url: http://localhost:5000/double
+          json:
+            number: !int '5'
+          method: POST
+          headers:
+            content-type: application/json
+        response:
+          status_code: 200
+          body:
+            double: !float 10
+    """)
+
+    return text
 
 
 class TestValidateFunctions:
@@ -201,3 +228,12 @@ class TestMatchRecursive:
         b["a"][0] = "wrong"
 
         check_keys_match_recursive(a, b, [])
+
+
+class TestCustomTokens:
+
+    def test_conversion(self, test_yaml):
+        stages = yaml.load(test_yaml, Loader=IncludeLoader)['stages'][0]
+
+        assert isinstance(stages['request']['json']['number'], int)
+        assert isinstance(stages['response']['body']['double'], float)

